airflow/models.py

```diff
@@ -895,6 +905,8 @@ def are_dependencies_met(
         if not main_session:
             session.commit()
             session.close()
+        if verbose:
+            logging.warning("Trigger rule `{}` not satisfied".format(tr))
```

Comment:
here in task._upstream_list we have a list of upstream dependencies, but if some is not satisfied we would still just see e.g. 'Trigger rule `ALL_SUCCESS` not satisfied' while airflow user would more likely be interested in the list of upstream tasks missing. This does not matter for ONE_SUCCESS and ONE_FAILED but for ALL_* it does.
Given the way the query is done now (using just sum of tasks in a specific state) it is not that straightforward to print it all out here, but you could consider that at some point.